native/src/routes/events/containers/EventsContainer.js

```diff
@@ -90,36 +90,62 @@ const mapStateToProps = (state: StateType, ownProps: OwnPropsType): StatePropsTy
   }
 
   if (state.cities.status === 'error') {
-    return { status: 'error', message: state.cities.message, code: state.cities.code, refreshProps }
+    return {
+      status: 'error',
+      message: state.cities.message,
+      code: state.cities.code,
+      refreshProps
+    }
   } else if (resourceCache.status === 'error') {
-    return { status: 'error', message: resourceCache.message, code: resourceCache.code, refreshProps }
+    return {
+      status: 'error',
+      message: resourceCache.message,
+      code: resourceCache.code,
+      refreshProps
+    }
   } else if (route.status === 'error') {
-    return { status: 'error', message: route.message, code: route.code, refreshProps }
+    return {
+      status: 'error',
+      message: route.message,
+      code: route.code,
+      refreshProps
+    }
   } else if (languages.status === 'error') {
-    return { status: 'error', message: languages.message, code: languages.code, refreshProps }
+    return {
+      status: 'error',
+      message: languages.message,
+      code: languages.code,
+      refreshProps
+    }
   }
 
   if (state.resourceCacheUrl === null || state.cities.status === 'loading' || switchingLanguage ||
     (route.status === 'loading' && !routeHasOldContent(route)) || languages.status === 'loading') {
     return { status: 'loading' }
   }
-
   const cities = state.cities.models
+  const innerProps = {
+    path: route.path,
+    events: route.models,
+    cities: cities,
+    cityCode: route.city,
+    language: route.language,
+    resourceCache: resourceCache.value,
+    resourceCacheUrl: state.resourceCacheUrl,
+    navigation
+  }
 
-  // $FlowFixMe Flow can't evaluate the status as it is dynamic
+  if (route.status === 'loading') {
```

Comment:
This is actually the [recommended solution from the flow config](https://flow.org/en/docs/lang/refinements/). 
To keep the one-liner we would need to redefine the StatusPropsType to have one status looking like the code below. Personally I like the if/else construct more
```javascript
export type StatusPropsType<S: {}, R: {}> =
   { status: 'loading' | 'success' | 'routeNotInitialized' | 'error' } &
  (RouteNotInitializedType
  | LoadingType<$Diff<S, { dispatch: Dispatch<StoreActionType> }>, R>
  | ErrorType<R>
  | LanguageNotAvailableType
  | SuccessType<$Diff<S, { dispatch: Dispatch<StoreActionType> }>, R>)
```